grakel/utils.py

```diff
@@ -679,3 +688,108 @@ def get_el(ef, g):
+            Input should be a torch_geometric data object to be converted.
+
+        node_one_hot : bool, default=False
+            Whether each element of `data.x` should be treated as a one-hot vector.
+
+        edge_one_hot : bool, default=False
+            Whether each element of `data.edge_attr` should be treated as a one-hot vector.
+
+        ignore_y : bool, default=False
+            Argument determining whether to return the labels connected to the input graphs.
+
+    Returns
+    -------
+        grakel_graphs : dict
+            Returns a dictionary with keys:
+
+                - 'graph' : `list[grakel.Graph]`
+                    
+                    + If input is torch_geometric.data.Data returns a single element.
+                    
+                    + If input is torch_geometric.data.Data returns a list of graphs.
+
+                - 'y' : `list[int]`
+                    
+                    + List of labels of graphs.
+                    
+                    + If ignore_y is True this argument is ommited.
+
+    """
+    def one_hot_node(x):
+        if node_one_hot:
+            return x.argmax().item()
+        else:
+            return x.cpu().numpy()
+
+    def one_hot_edge(x):
+        if edge_one_hot:
+            return x.argmax().item()
+        else:
+            return x.cpu().numpy()
+
+    if data.batch is None:
+        edges = [tuple(e) for e in data.edge_index.t().tolist()]
+        node_labels, edge_labels = [], []
+        if data.x is not None:
+            node_labels = {i: one_hot_node(data.x[i]) for i in range(data.x.shape[0])}
+
+        if data.edge_attr is not None:
+            edge_labels = {edges[i]: one_hot_edge(data.edge_attr[i]) for i in range(data.edge_attr.shape[0])}
+
+        x = {'graph': Graph(edges, node_labels, edge_labels)}
+        if data.y is not None:
+            try:
+                x['y'] = int(data.y.item())
+            except:
+                raise ValueError('Either set ignore_y=True or use a ')
+
+        return x
+    else:
+        edges = defaultdict(list)
+        node_labels = None
+        edge_labels = None if data.edge_attr is None else defaultdict(dict)
+        group_id = (data.batch).unique_consecutive().tolist()
+        lookup = {i: data.batch[i].item() for i in range(data.batch.shape[0])}
+        for i, e in enumerate(data.edge_index.t().tolist()):
+            e = tuple(e)
+            group = lookup[e[0]]
+            if lookup[e[0]] != lookup[e[1]]:
+                raise ValueError('Both nodes from an edge should correspond to the same graph.')
+            edges[group].append(e)
+            if  data.edge_attr is not None:
+                edge_labels[group][e] = one_hot_edge(data.edge_attr[i])
+
+        if data.x is not None:
+            node_labels = defaultdict(dict)
+            for i in range(data.x.shape[0]):
+                node_labels[data.batch[i].item()][i] = one_hot_node(data.x[i])
```

Comment:
This should be replaced:
`node_labels[data.batch[i].item()][i-cumsum_n_nodes[lookup[i]]] = one_hot_node(data.x[i])`